cirq/circuits/optimization_pass.py

```diff
@@ -41,8 +44,14 @@ def __init__(self, clear_span: int, clear_qubits: Iterable['cirq.Qid'],
                 with each affected moment.
             new_operations: The operations to replace the cleared out
                 operations with.
+            preserve_moments: Whether to keep Moments intact instead of
```

Comment:
To a user looking over this for the first time, this description is not very clear. I would rephrase into something like

```
If set, `cirq.Moment` instances within `new_operations` will be preserved exactly.
Normally the operations would be repacked to fit better into the target space, which
may move them between moments.
```

Reply:
you should update the description on `flatten_op_tree`, as I have replicated it here :) Using your wording plus the additional warning per kevin's concerns